src/databricks/labs/blueprint/installation.py

```diff
@@ -733,10 +797,23 @@ def _unmarshal_dict(cls, inst, path, type_ref):
     def _unmarshal_primitive(cls, inst, type_ref):
         """The `_unmarshal_primitive` method is a private method that is used to deserialize a dictionary to an object
         of type `type_ref`. This method is called by the `load` method."""
-        if not inst:
+        if inst is None:
+            return None
+        if isinstance(inst, type_ref):
             return inst
-        # convert from str to int if necessary
-        converted = type_ref(inst)  # type: ignore[call-arg]
+        converted = inst
+        # convert from str
+        if isinstance(inst, str):
+            if type_ref in (int, float):
+                try:
+                    converted = type_ref(inst)  # type: ignore[call-arg]
+                except ValueError as exc:
+                    raise SerdeError(f"Not a number {inst}!") from exc
+            elif type_ref == bool:
+                if inst.lower() == "true":
+                    converted = True
+                elif inst.lower() == "false":
+                    converted = False
```

Comment:
This introduces a behaviour change. Previously it would coerce values (eg. float/int/boolean) to strings if the type_ref was str, but now it doesn't. (This is why some tests needed updating, but there might be other code that relies on this.)

For compatibility purposes this needs to be fixed.

Reply:
There are some compatibility issues in UCX that are proving troublesome to resolve, in particular during tests some mocks are marshalled, and present in the raw data being unmarshalled. For now we're going to ignore that though.